tests/dialects/test_teradata.py

```diff
@@ -205,3 +205,41 @@ def test_cast(self):
                 "teradata": "TRYCAST('-2.5' AS DECIMAL(5, 2))",
             },
         )
+
+    def test_str_to_time(self):
+        self.validate_all(
+            "TO_DATE('2024-01-01', 'YYYY-MM-DD')",
+            write={
+                "teradata": "TO_DATE('2024-01-01', 'YYYY-MM-DD')",
+                "oracle": "TO_DATE('2024-01-01', 'YYYY-MM-DD')",
+                "postgres": "TO_DATE('2024-01-01', 'YYYY-MM-DD')",
+            },
+        )
+        self.validate_all(
+            "TO_TIMESTAMP('2024-01-01 13:14:15', 'YYYY-MM-DD HH24:MI:SS')",
+            write={
+                "teradata": "TO_TIMESTAMP('2024-01-01 13:14:15', 'YYYY-MM-DD HH24:MI:SS')",
+                "oracle": "TO_TIMESTAMP('2024-01-01 13:14:15', 'YYYY-MM-DD HH24:MI:SS')",
+                "postgres": "TO_TIMESTAMP('2024-01-01 13:14:15', 'YYYY-MM-DD HH24:MI:SS')",
+            },
+        )
+        self.validate_all(
+            "CAST('2024-01-01' AS DATE FORMAT 'YYYY-MM-DD')",
+            write={"teradata": "CAST('2024-01-01' AS DATE FORMAT 'YYYY-MM-DD')"},
+        )
```

Comment:
Nit: we generally try to use `validate_identity(input[, expected])` for tests like this where we're only interested to check the parsing / generation roundtrip for the given dialect, because it's more compact. So this would look like:

```python
        self.validate_identity("CAST('2024-01-01' AS DATE FORMAT 'YYYY-MM-DD')")
```